### ISSUE TITLE
Segfault on enabling async scheduled task

### ISSUE THREAD

**Issue by korthout** [camunda/zeebe#13164]
**Describe the bug**

A segfault can occur when one of the experimental feature flags is enabled and the other disabled:
- `ZEEBE_BROKER_EXPERIMENTAL_FEATURES_ENABLEMESSAGETTLCHECKERASYNC`
- `ZEEBE_BROKER_EXPERIMENTAL_FEATURES_ENABLETIMERDUEDATECHECKERASYNC`

This same error does not occur when both are enabled, or both are disabled.

**To Reproduce**

Enable either of, and disable the other:
- `ZEEBE_BROKER_EXPERIMENTAL_FEATURES_ENABLEMESSAGETTLCHECKERASYNC`
- `ZEEBE_BROKER_EXPERIMENTAL_FEATURES_ENABLETIMERDUEDATECHECKERASYNC`

**Expected behavior**

No segfault occurs.

**Log/Stacktrace**

<details><summary>Full Stacktrace</summary>
 <p>

```
Current thread (0x00007f4dbd280db0):  JavaThread "-zb-actors-7" [_thread_in_native, id=15747, stack(0x00007f4d1222d000,0x00007f4d1232e000)]

Stack: [0x00007f4d1222d000,0x00007f4d1232e000],  sp=0x00007f4d1232c340,  free space=1020k
Native frames: (J=compiled Java code, j=interpreted, Vv=VM code, C=native code)
C  [librocksdbjni14514272448621065586.so+0x5e2a98]  rocksdb::Arena::~Arena()+0x188
C  [librocksdbjni14514272448621065586.so+0x877aa6]  rocksdb::WriteBatchWithIndex::Rep::ClearIndex()+0x16
C  [librocksdbjni14514272448621065586.so+0x851e59]  rocksdb::TransactionBaseImpl::Clear()+0x349
C  [librocksdbjni14514272448621065586.so+0x83afca]  rocksdb::OptimisticTransaction::Rollback()+0x2a
C  [librocksdbjni14514272448621065586.so+0x2def2e]  Java_org_rocksdb_Transaction_rollback+0x1e
J 13789  org.rocksdb.Transaction.rollback(J)V (0 bytes) @ 0x00007f4dad4fa05b [0x00007f4dad4f9fa0+0x00000000000000bb]
J 14913 c1 org.rocksdb.Transaction.rollback()V (60 bytes) @ 0x00007f4da5bf2fbc [0x00007f4da5bf2e00+0x00000000000001bc]
J 14285 c2 io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.ensureInOpenTransaction(Lio/camunda/zeebe/db/impl/rocksdb/transaction/TransactionConsumer;)V (29 bytes) @ 0x00007f4dad398614 [0x00007f4dad398280+0x0000000000000394]
J 21410 c1 io.camunda.zeebe.db.impl.rocksdb.transaction.TransactionalColumnFamily.forEach(Ljava/util/function/BiConsumer;)V (24 bytes) @ 0x00007f4da5fb5a94 [0x00007f4da5fb56a0+0x00000000000003f4]
j  io.camunda.zeebe.engine.state.distribution.DbDistributionState.foreachPendingDistribution(Lio/camunda/zeebe/engine/state/immutable/DistributionState$PendingDistributionVisitor;)V+49
j  io.camunda.zeebe.engine.processing.distribution.CommandRedistributor.runRetryCycle()V+30
j  io.camunda.zeebe.engine.processing.distribution.CommandRedistributor$$Lambda$727+0x00000008015c5c08.run()V+4
J 18846 c1 io.camunda.zeebe.stream.api.scheduling.SimpleProcessingScheduleService.lambda$runAtFixedRate$0(Ljava/lang/Runnable;Ljava/time/Duration;)V (49 bytes) @ 0x00007f4da57d5274 [0x00007f4da57d5160+0x0000000000000114]
J 18845 c1 io.camunda.zeebe.stream.api.scheduling.SimpleProcessingScheduleService$$Lambda$720+0x00000008015c4698.run()V (18 bytes) @ 0x00007f4da5f4ecac [0x00007f4da5f4eb40+0x000000000000016c]
J 15127 c2 io.camunda.zeebe.scheduler.ActorJob.execute(Lio/camunda/zeebe/scheduler/ActorThread;)V (351 bytes) @ 0x00007f4dad5606ac [0x00007f4dad560540+0x000000000000016c]
J 18627 c2 io.camunda.zeebe.scheduler.ActorTask.execute(Lio/camunda/zeebe/scheduler/ActorThread;)Z (350 bytes) @ 0x00007f4dad8730f4 [0x00007f4dad872840+0x00000000000008b4]
J 22040 c2 io.camunda.zeebe.scheduler.ActorThread.doWork()V (285 bytes) @ 0x00007f4dadbb264c [0x00007f4dadbb2240+0x000000000000040c]
J 11375 c2 io.camunda.zeebe.scheduler.ActorThread.run()V (103 bytes) @ 0x00007f4dacf579e0 [0x00007f4dacf578e0+0x0000000000000100]
v  ~StubRoutines::call_stub
V  [libjvm.so+0x825045]  JavaCalls::call_helper(JavaValue*, methodHandle const&, JavaCallArguments*, JavaThread*)+0x315
V  [libjvm.so+0x82683b]  JavaCalls::call_virtual(JavaValue*, Handle, Klass*, Symbol*, Symbol*, JavaThread*)+0x1cb
V  [libjvm.so+0x8f14f3]  thread_entry(JavaThread*, JavaThread*)+0xa3
V  [libjvm.so+0xe62564]  JavaThread::thread_main_inner()+0x184
V  [libjvm.so+0xe65c10]  Thread::call_run()+0xc0
V  [libjvm.so+0xc1aa21]  thread_native_entry(Thread*)+0xe1
```

</p>
</details>

**Comment by korthout** [comment]
❗ Since merging #13008, this segfault can also occur when both feature flags are enabled. We should fix this soon! Luckily the fix is easy and already coming up.

The segfault occurs because of the following situation:
- the stream processor has its own transaction context
- the scheduled tasks have their own shared transaction context
- the scheduled tasks can be run on the stream processor's actor, or async on their scheduled tasks actor
- the problem occurs when two scheduled tasks run on different actors while reusing the shared transaction context

**Comment by korthout** [comment]
Since this issue only exists in a few places and those don't affect users, we'll look into it after the company retreat:
- on released versions (`8.2.7-` and `8.1.13-`, but not on any `8.3` alphas) when only one of these experimental feature flags are enabled and the other is disabled:
  - `ZEEBE_BROKER_EXPERIMENTAL_FEATURES_ENABLEMESSAGETTLCHECKERASYNC`
  - `ZEEBE_BROKER_EXPERIMENTAL_FEATURES_ENABLETIMERDUEDATECHECKERASYNC`
- on `main` branch when one or both feature flags are enabled
- on `main` branch in tests (because both feature flags are enabled)
